### ISSUE TITLE
Q&A: Is it possible to write a client code for FRR daemon using GoBGP library. 

### ISSUE THREAD

**Issue by sakthishanmugam02** [osrg/gobgp#2624]
Is it possible to write a client code for FRR daemon using GoBGP library. If possible, could anyone help, sharing references. I am not able to find the package `"github.com/osrg/gobgp/client"` based on the sample I got as below in the current version of GoBGP.

```
package main

import (
	"context"
	"fmt"
	"github.com/osrg/gobgp/client"
)

func main() {
	c, err := client.New("127.0.0.1:50051")
	if err != nil {
		fmt.Println("Error creating GoBGP client:", err)
		return
	}
	defer c.Close()

	routes, err := c.GetRib("", 0, false)
	if err != nil {
		fmt.Println("Error fetching routes:", err)
		return
	}

	for _, route := range routes {
		fmt.Println("Route:", route)
	}
}
```

**Comment by floatingstatic** [comment]
@sakthishanmugam02 No, instead you need to compile the proto definition provided by FRR into grpc client stubs in your language of choice using `protoc` and use that. See the proto definition here: https://github.com/opensourcerouting/frr/blob/master/grpc/frr-northbound.proto

Note, if you are using go the proto definition may need `option go_package` but you will have to raise that with the folks that maintain frr.

gobgp client is basically a cli/grpc client for gobgp's grpc service only implemented by gobgpd (server). FRR in turn has its own service definition and implementation so the client stub for gobgp is not compatible with FRR.
